py/dml/types.py

```diff
@@ -1146,13 +1169,21 @@ class TFunction(DMLType):
     def __init__(self, input_types, output_type,
                  varargs = False, const = False):
         assert isinstance(output_type, DMLType)
+        assert not self.const
```

Comment:
Why do this when you have
```
    @property
    def const(self): return False
```
straight-up clear as day? You might as well `assert True`. Especially as this does not protect against anything even _if_ that definition of `TFunction.const` would be removed as you're checking `self.const` before `DMLType.__init__` is called to initialize it. The check would be more meaningful if it was replaced with `assert not const`, or if it was moved to after `__init__` (but it would still be redundant regardless. Even the former check is covered by the `const.setter` override.)

Reply:
Right, leftover from before I realized the `@property` solution. Thanks for spotting.